go/pkg/yolosvc/testing.go

```diff
@@ -0,0 +1,178 @@
+package yolosvc
+
+import (
+	"fmt"
+	"testing"
+
+	"berty.tech/yolo/v2/go/pkg/testutil"
+	"berty.tech/yolo/v2/go/pkg/yolopb"
+	"berty.tech/yolo/v2/go/pkg/yolostore"
+	"github.com/jinzhu/gorm"
+	"go.uber.org/zap"
+)
+
+func TestingService(t *testing.T, opts ServiceOpts) (Service, func()) {
+	t.Helper()
+
+	opts.DevMode = true
+
+	if opts.Logger == nil {
+		opts.Logger = zap.NewNop()
+	}
+
+	db := testingDB(t)
+
+	api, err := NewService(db, opts)
+	if err != nil {
+		t.Fatalf("init api: %v", err)
+	}
+
+	cleanup := func() {
+		db.Close()
+	}
+
+	return api, cleanup
+}
+
+func testingDB(t *testing.T) *gorm.DB {
+	t.Helper()
+
+	db, err := gorm.Open("sqlite3", "file::memory:?cache=shared")
+	if err != nil {
+		t.Fatalf("init in-memory sqlite server: %v", err)
+	}
+
+	logger := testutil.Logger(t)
+
+	db, err = yolostore.InitDB(db, logger)
+	if err != nil {
+		t.Fatalf("init in-memory db: %v", err)
+	}
+
+	testingCreateEntities(t, db)
+
+	return db
+}
+
+func testingCreateEntities(t *testing.T, db *gorm.DB) {
+	t.Helper()
+
+	if err := db.Transaction(func(tx *gorm.DB) error {
```

Comment:
```suggestion
	err := db.Transaction(func(tx *gorm.DB) error {
```

and at the end, a standalone `if err != nil` (it's a good habit I start taking to avoid inlined `if err := STUFF; err != nil` when STUFF is more than one or few lines